lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2538,6 +2552,31 @@ auto ClangImporter::Implementation::importFullName(
 
   // Omit needless words.
   StringScratchSpace omitNeedlessWordsScratch;
+
+  // swift_newtype-ed declarations may have common words with the type name
+  // stripped.
+  if (auto newtypeDecl = findSwiftNewtype(D)) {
+    SmallString<32> newNameBuffer;
+    auto newtypeNameWords = camel_case::getWords(newtypeDecl->getName());
+    auto declNameWords = camel_case::getWords(baseName);
+
+    // Scan the decl name, and drop any words that appear in order in the type
+    // name
+    for (auto wI = declNameWords.begin(), skipWI = newtypeNameWords.begin();
```

Comment:
Not in the spirit of SE-0033, in which there's the example:
extern NSErrorDomain const NSCocoaErrorDomain ===> NSErrorDomain.cocoa

In this case, it's not prefix stripping but rather common-word stripping. I'm happy to make a utility in StringExtras or wherever that can help expose, share, or unify these tasks, but I wasn't able to find a direct analogue to what I was trying to do. 

In general, I am interested in more camel_case utilities that can also help unify with the import-as-member inference system (which has its own handful of helpers) and enum handling (which is just prefix).